osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            foreach (var i in items)
+            {
+                ((Drawable)i).Expire();
+                SelectedItems.Remove(i);
+            }
+        }
+
+        protected override IEnumerable<MenuItem> GetContextMenuItemsForSelection(IEnumerable<SelectionBlueprint<ISkinnableComponent>> selection)
+        {
+            yield return new OsuMenuItem("Anchor")
+            {
+                Items = createAnchorItems().ToArray()
+            };
+
+            foreach (var item in base.GetContextMenuItemsForSelection(selection))
+                yield return item;
+
+            IEnumerable<AnchorMenuItem> createAnchorItems()
+            {
+                var displayableAnchors = new[]
+                {
+                    Anchor.TopLeft,
+                    Anchor.TopCentre,
+                    Anchor.TopRight,
+                    Anchor.CentreLeft,
+                    Anchor.Centre,
+                    Anchor.CentreRight,
+                    Anchor.BottomLeft,
+                    Anchor.BottomCentre,
+                    Anchor.BottomRight,
+                };
+
+                return displayableAnchors.Select(a =>
+                {
+                    var countExisting = selection.Count(b => ((Drawable)b.Item).Anchor == a);
+                    var countTotal = selection.Count();
+
+                    TernaryState state;
+
+                    if (countExisting == countTotal)
+                        state = TernaryState.True;
+                    else if (countExisting > 0)
+                        state = TernaryState.Indeterminate;
+                    else
+                        state = TernaryState.False;
+
+                    return new AnchorMenuItem(a, selection, _ => applyAnchor(a))
+                    {
+                        State = { Value = state }
+                    };
+                });
+            }
+        }
+
+        private void applyAnchor(Anchor anchor)
+        {
+            foreach (var item in SelectedItems)
+                ((Drawable)item).Anchor = anchor;
+        }
+
+        protected override void OnSelectionChanged()
+        {
+            base.OnSelectionChanged();
+
+            SelectionBox.CanRotate = true;
+            SelectionBox.CanScaleX = true;
+            SelectionBox.CanScaleY = true;
+            SelectionBox.CanReverse = false;
+        }
+
+        public override bool HandleRotation(float angle)
+        {
+            // TODO: this doesn't correctly account for origin/anchor specs being different in a multi-selection.
+            foreach (var c in SelectedBlueprints)
+                ((Drawable)c.Item).Rotation += angle;
+
+            return base.HandleRotation(angle);
+        }
+
+        public override bool HandleScale(Vector2 scale, Anchor anchor)
+        {
+            adjustScaleFromAnchor(ref scale, anchor);
+
+            foreach (var c in SelectedBlueprints)
+                ((Drawable)c.Item).Scale += scale * 0.01f;
```

Comment:
This `0.01f` number is temporary/arbitrary, right? Doesn't feel right currently. The drag box handles are definitely in a different place than the mouse after a drag.

Reply:
I think I took this from the existing implementation. All the scale/rotation logic needs a complete reimplementation though, as it doesn't actually do what it needs to (see the beatmap editor one for a better example of what needs to happen).

Reply:
I've adjusted the multiplier slightly and also added an inline comment mentioning that this will be fixed in a separate pass.